crates/whiskers/src/grid/mod.rs

```diff
@@ -0,0 +1,181 @@
+                        None => None,
+                    },
+                    size: self.module_size(),
+                });
+            }
+        }
+
+        self.data = data;
+        self
+    }
+
+    /// Returns optional reference to a grid cell at specific column and row index
+    pub fn at(&mut self, column: usize, row: usize) -> Option<&mut GridCell<T>> {
+        self.data
+            .iter_mut()
+            .find(|cell| cell.column == column && cell.row == row)
+    }
+
+    /// Resets grid cell's data at specific column and row if applicable
+    pub fn reset_at(&mut self, column: usize, row: usize) {
+        let cell = self.at(column, row);
+
+        if let Some(cell) = cell {
+            cell.reset_data();
+        }
+    }
+
+    /// Resets all grid cell's data
+    pub fn reset(&mut self) {
+        self.data.iter_mut().for_each(GridCell::reset_data);
+    }
+
+    /// Aligns grid cells to specific dimensions
+    pub fn align_to(&mut self, rect_dimensions: [f64; 2]) {
+        let [width, height] = self.module_size();
+        let diff_x = rect_dimensions[0] / width;
+        let diff_y = rect_dimensions[1] / height;
+
+        self.data.iter_mut().for_each(|cell| {
+            cell.canvas_position = Point::new(
+                cell.canvas_position.x() + diff_x,
+                cell.canvas_position.y() + diff_y,
+            );
+        });
+    }
+
+    /// Returns width of the grid
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn width(&self) -> f64 {
+        self.columns as f64 * self.module_size()[0] + self.columns as f64 * self.gutter[0]
+    }
+
+    /// Returns height of the grid
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn height(&self) -> f64 {
+        self.rows as f64 * self.module_size()[1] + self.rows as f64 * self.gutter[1]
+    }
+
+    /// Returns grid module's size
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn module_size(&self) -> [f64; 2] {
+        match self.size {
+            GridSize::GridBased([width, height]) => {
+                let cols = self.columns as f64;
+                let rows = self.rows as f64;
+                let [gutter_width, gutter_height] = self.gutter;
+
+                [
+                    (width - (cols - 1.0) * gutter_width) / cols,
+                    (height - (rows - 1.0) * gutter_height) / rows,
+                ]
+            }
+            GridSize::CellBased(s) => s,
+        }
+    }
+
+    /// Draw grid's instance as rectangles, with current sketch settings
+    pub fn draw(&mut self) {
+        self.data.iter().for_each(|cell| {
+            self.sketch.rect(
+                cell.canvas_position.x() + (cell.size[0] / 2.0),
+                cell.canvas_position.y() + (cell.size[1] / 2.0),
+                cell.size[0],
+                cell.size[1],
+            );
+        });
+    }
```

Comment:
That's a good idea, but I'd make `Cell` a `IntoBezPathTolerance` instead, so that we can do that:

```rust
Grid::from_cell_size(...)
    //...
    .build(sketch, |sketch, cell| {
        sketch.add_path(cell); // draws boundaries
    });
```